src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -1120,9 +1109,9 @@ private static String federatedFlowStep4(
         logger.debug("The specified authenticator {} and the destination URL " +
                      "in the SAML assertion {} do not match.",
                      loginInput.getAuthenticator(), postBackUrl);
-        throw new SnowflakeSQLException(
+        throw new SnowflakeSQLLoggedException(
             SqlState.SQLCLIENT_UNABLE_TO_ESTABLISH_SQLCONNECTION,
-            ErrorCode.IDP_INCORRECT_DESTINATION.getMessageCode());
+            ErrorCode.IDP_INCORRECT_DESTINATION.getMessageCode(), /* session = */ null);
```

Comment:
Not a good way to get a session for this function.

Reply:
Can you add your comment in the code?